gravitee-apim-console-webui/src/management/settings/groups/group/group.component.ts

```diff
@@ -13,32 +13,700 @@
+      this.defaultAPIRoles = roles.sort((a, b) => a.name.localeCompare(b.name));
+    });
+
+    this.roleService.list('APPLICATION').subscribe((roles) => {
+      this.defaultApplicationRoles = roles.sort((a, b) => a.name.localeCompare(b.name));
+    });
+
+    this.roleService.list('INTEGRATION').subscribe((roles) => {
+      this.defaultIntegrationRoles = roles.sort((a, b) => a.name.localeCompare(b.name));
+    });
+  }
+
+  saveOrUpdate() {
+    this.updateEventRules();
+    this.updateRoles();
+
+    this.groupService
+      .saveOrUpdate(this.mode, this.mapUpdatedGroup())
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe({
+        next: (group) => {
+          this.snackBarService.success(`Successfully saved the group.`);
+
+          if (this.mode === 'new') {
+            this.router.navigate(['..', group.id], { relativeTo: this.route });
+          } else {
+            this.refreshGroup.next(1);
+          }
+        },
+        error: () => this.snackBarService.error(`Error occurred while saving the group.`),
+      });
+  }
+
+  openDeleteMemberDialog(member: Member) {
+    this.matDialog
+      .open<DeleteMemberDialogComponent, DeleteMemberDialogData, DeleteMemberDialogResult>(DeleteMemberDialogComponent, {
+        data: { member, members: this.groupMembers.value },
+        role: 'alertdialog',
+        id: 'deleteMemberDialog',
+        hasBackdrop: true,
+        autoFocus: true,
+        width: GIO_DIALOG_WIDTH.SMALL,
+      })
+      .afterClosed()
+      .pipe(
+        filter((dialogResult: DeleteMemberDialogResult) => dialogResult.shouldDelete),
+        mergeMap((dialogResult: DeleteMemberDialogResult) => {
+          return this.groupService.deleteMember(this.groupId, member.id).pipe(
+            tap(() => {
+              this.snackBarService.success('Successfully deleted member from the group.');
+              this.refreshGroupMembers.next(1);
+            }),
+            catchError(() => {
+              this.snackBarService.error('Error occurred while deleting member from the group.');
+              return EMPTY;
+            }),
+            map(() => dialogResult),
+          );
+        }),
+        switchMap((dialogResult: DeleteMemberDialogResult) => {
+          const shouldTransferOwnership = member.roles['API'] === RoleName.PRIMARY_OWNER;
+
+          return shouldTransferOwnership
+            ? this.groupService.addOrUpdateMemberships(this.groupId, [dialogResult.primaryOwnerMembership]).pipe(
+                tap(() => {
+                  this.snackBarService.success('Successfully transferred the ownership.');
+                  this.refreshGroupMembers.next(1);
+                }),
+                catchError(() => {
+                  this.snackBarService.error('Error occurred while transferring the ownership.');
+                  return EMPTY;
+                }),
+              )
+            : of(null);
+        }),
```

Comment:
```suggestion
        filter((_) => member.roles['API'] === RoleName.PRIMARY_OWNER),
        switchMap((dialogResult: DeleteMemberDialogResult) => 
            this.groupService.addOrUpdateMemberships(this.groupId, [dialogResult.primaryOwnerMembership]).pipe(
                tap(() => {
                  this.snackBarService.success('Successfully transferred the ownership.');
                  this.refreshGroupMembers.next(1);
                }),
                catchError(() => {
                  this.snackBarService.error('Error occurred while transferring the ownership.');
                  return EMPTY;
                }),
              )
        ),
```
I was finding it a bit hard to read... What do you think about extracting the `shouldTransferOwnership` to a filter before the switchMap?